nexus/src/authz/api_resources.rs

```diff
@@ -19,13 +19,87 @@
 // pass them to `is_allowed()`.  Using newtypes is a way to capture just the
 // parts we need for authorization.
 
-use super::actor::AuthenticatedActor;
-use super::roles::AuthzApiResource;
-use super::roles::AuthzResource;
+use super::actor::AnyActor;
+use super::context::Authorize;
+use super::roles::{
+    load_roles_for_resource, load_roles_for_resource_tree, RoleSet,
+};
+use super::Action;
+use super::{actor::AuthenticatedActor, Authz};
+use crate::authn;
+use crate::context::OpContext;
 use crate::db::fixed_data::FLEET_ID;
-use omicron_common::api::external::ResourceType;
+use crate::db::DataStore;
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use omicron_common::api::external::{Error, LookupType, ResourceType};
 use uuid::Uuid;
 
+/// Describes an authz resource that corresponds to an API resource that has a
+/// corresponding ResourceType and is stored in the database
+///
+/// This is a helper trait used to impl [`AuthzResource`].
+pub trait AuthzApiResource: Clone + Send + Sync + 'static {
```

Comment:
This trait moved from nexus/src/authz/roles.rs because it grew a new function (`not_found`) which is not really related to roles.  On the plus side, the trait and its implementors are all in this file now.

Reply:
Fixed in484ae6bcf0901fa55cd6c5142edb40ece4bca9e1.